Gems/Atom/RHI/Code/Include/Atom/RHI/CpuProfiler.h

```diff
@@ -20,25 +21,21 @@ namespace AZ
         //! Structure that is used to cache a timed region into the thread's local storage.
         struct CachedTimeRegion
         {
-            //! Structure that the profiling macro utilizes to create statically initialized instance to create string
-            //! literals in static memory
-            struct GroupRegionName
-            {
-                GroupRegionName() = delete;
-                GroupRegionName(const char* const group, const char* const region);
-                
-                const char* const m_groupName = nullptr;
-                const char* const m_regionName = nullptr;
-            };
-
             CachedTimeRegion() = default;
-            CachedTimeRegion(const GroupRegionName* groupRegionName);
-            CachedTimeRegion(const GroupRegionName* groupRegionName, uint16_t stackDepth, uint64_t startTick, uint64_t endTick);
+            CachedTimeRegion(const char* const groupName, AZStd::string_view regionName);
+            CachedTimeRegion(AZStd::string_view groupName, Name regionName, uint16_t stackDepth, uint64_t startTick, uint64_t endTick);
+
+            ~CachedTimeRegion() = default;
+            CachedTimeRegion(const CachedTimeRegion& other) = default;
+            CachedTimeRegion(CachedTimeRegion&& other) noexcept = default;
+            CachedTimeRegion& operator=(const CachedTimeRegion& rhs) = default;
+            CachedTimeRegion& operator=(CachedTimeRegion&& rhs) noexcept;
```

Comment:
I'm going to bring this up in ly-engineers tomorrow, but there's some strange behavior with `Name`'s move assignment operator `Name& operator=(Name&& rhs)` that this is a hack to get around. When I leave the default move assignment in, this code hits an assert in `SetEmptyString`. Here's what I understood from a few hours of debugging:

ImGuiCpuProfiler.inl line 529 we call `vec.erase` on a vector of CachedTimeRegions, which have a `Name` member. This call internally moves the CachedTimeRegions with `operator=(CachedTimeRegion&& rhs)` to account for erases. All other assignments are fine except for the Name move assignment (Name.cpp line 62) which asserts. Going deeper, the issue is centered around line 76 - we are moving an `intrusive_ptr` which should null out the pointer. However, when we check `rhs.SetEmptyString()`, the pointer is not nulled out, hitting the assert. I'm thinking this has to be some sneaky UB somewhere in this change, since I doubt a bug like this would have stayed in the name system for so long. 

I can put in a FIXME here and leave it worst comes to worst.

Reply:
After rebasing onto latest development this issue seems to be fixed after the call was switched to `erase_if` from PR #2779 .